src/main/java/picard/analysis/directed/RnaSeqMetricsCollector.java

```diff
@@ -38,6 +38,7 @@
     final private Set<Integer> ignoredSequenceIndices;
 
     private final OverlapDetector<Gene> geneOverlapDetector;
+    private final Set<Gene> allGenes;
```

Comment:
You could declare this as `Collection<>` to avoid an issue if the return type of `OverlapDetector.getAll()` ever changes, since your code doesn't care if it's a `Set` or not.